src/sql/src/plan/statement/ddl.rs

```diff
@@ -1814,18 +1814,45 @@ pub fn plan_create_materialized_view(
 
 #[allow(clippy::too_many_arguments)]
 fn kafka_sink_builder(
-    _scx: &StatementContext,
+    scx: &StatementContext,
+    connection: mz_sql_parser::ast::KafkaConnection<Aug>,
     format: Option<Format<Aug>>,
     consistency: Option<KafkaConsistency<Aug>>,
     with_options: &mut BTreeMap<String, SqlValueOrSecret>,
-    broker: String,
     topic_prefix: String,
     relation_key_indices: Option<Vec<usize>>,
     key_desc_and_indices: Option<(RelationDesc, Vec<usize>)>,
     value_desc: RelationDesc,
     envelope: SinkEnvelope,
     topic_suffix_nonce: String,
 ) -> Result<StorageSinkConnectionBuilder, PlanError> {
+    let config_options = kafka_util::extract_config(with_options)?;
+
+    let connection = match connection {
+        mz_sql_parser::ast::KafkaConnection::Reference { connection } => {
+            let item = scx.get_item_by_resolved_name(&connection)?;
+            // Get Kafka connection
+            let connection = match item.connection()? {
+                Connection::Kafka(connection) => connection.clone(),
+                _ => sql_bail!("{} is not a kafka connection", item.name()),
+            };
+            // TODO: Once we remove use of `String`-keyed options, we
+            // can remove this check because permitted values here will
+            // simply be a disjoint set of what `CONNECTION`s support.
+            for k in BTreeMap::<String, StringOrSecret>::from(connection.clone()).keys() {
+                if config_options.contains_key(k) {
+                    sql_bail!(
+                        "cannot set option {} for SINK using CONNECTION {}",
+                        k,
+                        scx.catalog.resolve_full_name(item.name())
+                    );
+                }
+            }
```

Comment:
is there a new test for this? I'm having trouble wrapping my head around what its doing

Reply:
I did not add a test for this, but it is the same condition as:
https://github.com/MaterializeInc/materialize/blob/main/test/kafka-ssl/smoketest.td#L207-L221